File: test/page_objects/admin/checkSettingsPage.js
/* global element  by expect browser */

class CheckSettingsPage {
  constructor () {
    this.questionTimeLimit = element(by.css('#questionTimeLimit'))
    this.timeBetweenQuestions = element(by.css('#loadingTimeLimit'))
    this.saveChanges = element(by.css('[value=\'Save changes\']'))
    this.errorMessage = element(by.css('.error-message'))
    this.errorSummaryMessage = element(by.css('.error-summary-list'))
    this.errorSummary = element(by.css('.error-summary'))
    this.successfulChanges = element(by.css('.box-successful h2'))
  }

  updateQuestionTimeLimit (value) {
    this.questionTimeLimit.clear()
    this.questionTimeLimit.sendKeys(value)
    this.saveChanges.click()
  }
}

module.exports = new CheckSettingsPage()

File: admin/test/features/support/page_objects/check_settings_page.rb
class CheckSettingsPage < SitePrism::Page
  set_url '/service-manager/check-settings'

  element :question_time_limit, "#questionTimeLimit"
  element :loading_time_limit, "#loadingTimeLimit"
  element :save_changes, "input[value='Save']"
  element :cancel, "input[value='Cancel']"
  element :error_message, '.error-message'
  element :error_summary_list, '.error-summary-list'
  element :error_summary, '.error-summary p'
  element :successful_submission, '.box-successful h2'

  def update_question_time_limit(value)
    question_time_limit.set ''
    question_time_limit.set value
    save_changes.click
  end

  def update_loading_time_limit(value)
    loading_time_limit.set ''
    loading_time_limit.set value
    save_changes.click
  end
end
